### ISSUE TITLE
Ceases to read after a while, hangs

### ISSUE THREAD

**Issue by bkenn011** [cterwilliger/VeDirectFrameHandler#6]
Anyone else experienced this?  I was able to get it to run fine on a Particle Photon using both hardware serial and software serial, the latter as I needed the one free hw serial to do something else a little more intensive.

Seems like after somewhere between a minute and 5 minutes it'll just cease to read and my output to the arduino IDE serial monitor quits entirely.

**Comment by cterwilliger** [comment]
Yes, I have seen that. I have two devices, a Victron BMV-712 and a Victron SmartSolar MPPT-150/85. The BMV-712 has worked flawlessly with an ESP8266.  The MPPT saw the same issues you saw with an ESP8266 and the exact same code. I changed it to an ESP32 and the issue went away. I use only HW serial. SW serial has problems keeping up.

**Comment by bkenn011** [comment]
Interesting, but glad to know I'm not the only one.  This library manages to provide reliable data (kudos) until it goes kaput.  Not the case with the one I cobbled together a few years ago, where rogue data appears over hardware and software serial alike.  I solved the issue on that project by incorporating a nano and a particle photon, where the nano can process and validate the MPPT data, repackage to something more easily parsed, and then sent to the photon at a slower rate (and read via software serial).  Clunky, but hasn't given me any problems.  I'm guessing you used a logic shifter for the MPPT->ESP32 to handle the 5v?

**Comment by cterwilliger** [comment]
I use the library to grab the data, JSONize it, and send it to a MQTT broker.  So, the ESP8266 or ESP32 don’t spend much time handling the data.  Not sure why 8266 can’t handle the stream from MPPT, but ESP32 works.  Yes, using a simple resistor voltage divider to convert 5V to 3V.

**Comment by cterwilliger** [comment]
Please try with with new VeDirectFrameHandler.h.  frameLen changed from 18 to 22 per new protocol doc.

**Comment by I-Connect** [comment]
not sure if your issue is fixed, I had same sort of issue (also using software serial) and was able to resolve it

https://github.com/cterwilliger/VeDirectFrameHandler/issues/11

**Comment by DannyAt** [comment]
> not sure if your issue is fixed, I had same sort of issue (also using software serial) and was able to resolve it
> 
> #11

How did you resolve it, I have same issue?

**Comment by giacinti** [comment]
Not sure if it may be related but hex protocol can somehow interfere as unhandled packets are regularly sent which causes the main loop to break with a bad crc... I have proposed a PR for implementing hex protocol https://github.com/cterwilliger/VeDirectFrameHandler/pull/7 and using this version flawlessly since then

**Comment by cterwilliger** [comment]
I've been on the road all summer with spotty internet.  Let me look at PR #7 and I'll see if we can get that merged as well as some other possible fixes. Philippe, how did you test the PR?

**Comment by giacinti** [comment]
I'm currently using the version with hex protocol on an ESP32 device connected to a Victron MPPT 75/15 with a very old firmware (139). I have implemented an ESPHOME component leveraging this library and use MQTT for gathering data and sending hex command (see here: https://github.com/giacinti/vedirect)
